static/js/controllers/contacts-content.js

```diff
@@ -150,7 +153,10 @@ var _ = require('underscore'),
         return split[0].concat(split[1]);
       };
 
-      var isContactPrimaryContact = function(contactDoc) {
+      var isPersonAndPrimaryContact = function(contactDoc) {
+        if (contactDoc.type !== 'person') {
```

Comment:
The problem was I couldn't parallelize it, ie: you can't say `$q.all([ if x then y, ... ]);`. So in this case it was a pragmatic problem not simply a design preference. Initially I added a second function to do `if (contactDoc.type !== 'person') return $q.resolve(); return isContactPrimaryContact();` but it looked really dumb and it doesn't actually solve you problem - you still have a function that branches on user type.